src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -3524,6 +3538,121 @@ public static void parseTownUnclaimCommand(Player player, String[] split) {
 		}
 	}
 
+	public static void parseTownMergeCommand(Player player, String[] args) {
+		Resident resident = TownyUniverse.getInstance().getResident(player.getUniqueId());
+		Town succumbingTown = TownyUniverse.getInstance().getTown(args[0]);
+		Town remainingTown = null;
+		try {
+			remainingTown = resident.getTown();
+		} catch (NotRegisteredException ignored) {}
+
+		// A lot of checks.
+		if (succumbingTown == null || succumbingTown.getName().equals(remainingTown.getName())) {
+			TownyMessaging.sendErrorMsg(player, Translation.of("msg_err_invalid_name", args[0]));
+			return;
+		}
+		
+		if (TownySettings.getMaxDistanceForTownMerge() > 0 && homeBlockDistance(remainingTown, succumbingTown) > TownySettings.getMaxDistanceForTownMerge()) {
+			TownyMessaging.sendErrorMsg(player, Translation.of("msg_town_merge_err_not_close", succumbingTown.getName(), TownySettings.getMaxDistanceForTownMerge()));
+			return;
+		}
+
+		int newResidentsAmount = remainingTown.getResidents().size() + succumbingTown.getResidents().size();
+		if (TownySettings.getMaxResidentsPerTown() > 0 && newResidentsAmount > TownySettings.getMaxResidentsPerTown()) {
+			TownyMessaging.sendErrorMsg(player, Translation.of("msg_town_merge_err_too_many_residents", TownySettings.getMaxResidentsPerTown()));
+			return;
+		}
+		
+		if ((remainingTown.getTownBlocks().size() + succumbingTown.getTownBlocks().size()) > TownySettings.getMaxTownBlocks(remainingTown, newResidentsAmount)) {
+			TownyMessaging.sendErrorMsg(player, Translation.of("msg_town_merge_err_too_many_townblocks", TownySettings.getMaxTownBlocks(remainingTown, newResidentsAmount)));
+			return;
+		}
+
+		if (TownySettings.isAllowingOutposts() && (remainingTown.getMaxOutpostSpawn() + succumbingTown.getMaxOutpostSpawn()) > remainingTown.getOutpostLimit()) {
+			TownyMessaging.sendErrorMsg(player, Translation.of("msg_town_merge_err_too_many_outposts", remainingTown.getOutpostLimit()));
+			return;
+		}
+
+		if (!BukkitTools.isOnline(succumbingTown.getMayor().getName()) || succumbingTown.getMayor().isNPC()) {
+			TownyMessaging.sendErrorMsg(player, Translation.of("msg_town_merge_other_offline", succumbingTown.getName(), succumbingTown.getMayor().getName()));
+			return;
+		}
+

```

Comment:
Something should probably be done right about here where we take into account if the succumbing town is in debt, and if so, warn the remaining town it will pay for the debt.